Mathlib/Order/SuccPred/CompleteLinearOrder.lean

```diff
@@ -59,6 +59,34 @@ lemma IsGLB.exists_of_nonempty_of_not_isPredLimit
     (hf : IsGLB (Set.range f) x) (hx : ¬ IsPredLimit x) :
     ∃ i, f i = x := hf.mem_of_nonempty_of_not_isPredLimit (Set.range_nonempty f) hx
 
+open Classical in
+/-- Every conditionally complete linear order with well-founded `<` is a successor order, by setting
+the succesor of an element to the infimum of all larger elements. -/
+noncomputable def ConditionallyCompleteLinearOrder.toSuccOrder [WellFoundedLT α] :
+    SuccOrder α where
+  succ a := if IsMax a then a else sInf {b | a < b}
+  le_succ a := by
+    by_cases h : IsMax a
+    · simp [h]
+    · simp only [h, ↓reduceIte]
+      rw [not_isMax_iff] at h
+      exact le_csInf h (fun b => le_of_lt)
+  max_of_succ_le hs := by
+    by_contra h
+    simp [h] at hs
+    rw [not_isMax_iff] at h
+    exact hs.not_lt (csInf_mem h)
+  le_of_lt_succ := by
+    intro b a hs
+    by_cases h : IsMax a
+    · simpa [h] using le_of_lt hs
+    · simp [h] at hs
+      simpa using not_mem_of_lt_csInf hs ⟨a, fun _ hc => hc.le⟩
+  succ_le_of_lt := by
+    intro a b ha
```

Comment:
```suggestion
  succ_le_of_lt a b ha := by
```

Reply:
This doesn't work, unfortunately. I think implicit variables might be the issue.

Reply:
```suggestion
  succ_le_of_lt {a b} ha := by
```
then